### ISSUE TITLE
Request to discuss maintenance cowardice

### ISSUE THREAD

**Issue by benwiley4000** [lerna/lerna#1635]
Frankly the license revert and removal of Jamie from the maintainers list displays cowardice. Your actions state, "we want companies collaborating with ICE to be allowed to profit from our open source software."

Which is your choice, but stop giving in to the self-righteous trolls claiming politics can be separated from software. It's inextricably linked, even though we try to ignore it routinely as an industry.

And for anyone wondering ... The original change was [approved by other maintainers](https://github.com/lerna/lerna/pull/1616), but then Jamie was removed from the repo for a code of conduct violation in response to pressure from outside.

*note:* the series of comments below were part of a discussion with another commenter, who since deleted their own. I edited in comments copied from my notification emails, to add context.

**Comment by benwiley4000** [comment]
I am of the opinion that the license could have been reviewed more but the decision to simply revert to MIT and remove Jamie from the project sends the signal that you will give in to political pressure as long as it comes disguised as centrism.

**Comment by robbyoconnor** [comment]
The license is invalid.

**Comment by benwiley4000** [comment]
I'm not going to stand idly by while open source community members weep crocodile tears for the welfare of corporations who have been "personally attacked" or "harassed" for their government contracts. These are entirely avoidable business decisions that should bear consequences when they veer into unjustifiable territory.

**Comment by benwiley4000** [comment]
Where do we draw the line? Is IBM also the victim of personal attacks when it is called out for its computer assistance to Nazi Germany?

**Comment by BillDinger** [comment]
The CoC explicitly stated they will not discriminate based on politics. That means atrocious politics not just whatever arbitrary line you are drawing.

If you wish to amend the CoC to say something else, like can discriminate against no one but Nazis or something go open a PR. But as written what he did was wrong

**Comment by benwiley4000** [comment]
Unclear why the commenter deleted everything they wrote - now I look like I'm talking to myself! 😄

**Comment by threecity** [comment]
Two of the Big 4 financial firms were on the list, I work at one that wasn't. I went to work today and insisted we do what we can to stay off this list. I was ready to fight for it, and I was willing to lose if it came to it, an honorable casualty.

You almost started something.

**Comment by catcher-in-the-try** [comment]
You may want to close this issue and get off the news for a few days.

**Comment by ThePerminator** [comment]
"Software is inextricably political" I don't get how it is?

Jamie should seek help he seems to be dealing with depression or something and is acting out because of it

**Comment by catcher-in-the-try** [comment]
@ThePerminator Do not make ad hominem arguments even if you disagree.

**Comment by tbredin** [comment]
Because of this, its sparked a discussion and we're currently investigating how to move to a "no harm" license like this: 
https://github.com/raisely/NoHarm

**Comment by benwiley4000** [comment]
> "Software is inextricably political" I don't get how it is?

Good question!

I think open source software is inherently political for a few reasons:
1. The origins of the free/open source software movement are super political which is apparent if you read any book/reference on the history of it
2. The technology we build is used to enable new capabilities for the organizations that use it, which means we enable policy enacted by these organizations.
3. Open source software today is used to power an overwhelming portion of the code run by private companies including those listed in the exclusionary license. By making code available to those organizations, we are enabling them.

> Jamie should seek help he seems to be dealing with depression or something and is acting out because of it

I have no idea if Jamie deals with depression but it's irrelevant - clearly Jamie was echoing a commonly held political sentiment about ICE. Please don't diminish that by trying to suggest he's mentally unstable.

**Comment by ThePerminator** [comment]
My intention is not to attack him, I'm worried about him

**Comment by benwiley4000** [comment]
If you're genuinely worried about Jamie you can reach out to him personally on Twitter or elsewhere instead of writing comments to discredit the legitimacy of his actions on GitHub threads.

**Comment by TheLarkInn** [comment]
Ben I'm going to close this issue, in favor or a more structured, RFC style of a proposal for what decisions you are questioning. Thanks!

**Comment by benwiley4000** [comment]
Please link to whatever new thread you create if you're going to close this - thanks.

**Comment by ThePerminator** [comment]
I am, which is why he has me worried what with his threats 

The legitimacy of his actions are not a matter of discussion as the maintainers of this project showed he breached the code of conduct and his erratic behaviour has brought a bad reputation to our community. There's no debating that he did the wrong thing. 
Attacking the maintainers and calling them "cowards" is also unacceptable, you should be ashamed of yourself

**Comment by krainboltgreene** [comment]
@TheLarkInn Please do link to the new thread you create.

**Comment by benwiley4000** [comment]
Closing the issue under the pretense of opening a new thread and then failing to do so says you just want to make everyone shut up.

**Comment by RyanCavanaugh** [comment]
@benwiley4000 Sean's last comment was at 1 AM local time. Please **assume good faith**, let people sleep, and allow time for reflection. There's nothing to be rushed here and accusing him of "making everyone shut up" because he didn't pull an all-nighter to post a proper discussion thread isn't civil.

**Comment by krainboltgreene** [comment]
Everyone's being civil @RyanCavanaugh, please assume good faith.

**Comment by benwiley4000** [comment]
I'm suggesting you shouldn't close an issue if it's not resolved or replaced.

**Comment by hannahhoward** [comment]
Till then... https://github.com/lerna/lerna/issues/1639

**Comment by RyanCavanaugh** [comment]
If someone says they're going to do something, then you should assume they're going to do that thing, and are doing it for earnest reasons. That's what good faith looks like.

If someone does something which is facially reasonable, and you accuse them for doing it under false and malicious pretenses, that is bad faith.

If Sean says he's going to open a discussion thread, then I trust that that will happen in due time. Whether this issue is open or closed *means nothing*; it is literally a bit in a database somewhere. Not having an database row with a bit set to zero for some particular thing for 2 days is not evidence that someone is trying to shut you up.

**Comment by krainboltgreene** [comment]
Please try to be civil @RyanCavanaugh, we're all trying to discuss this and it would be nice if you assumed good faith in everyone.

**Comment by benwiley4000** [comment]
Ryan I'm not going to get into a debate about "civility" but I'm saying that if Sean wasn't ready to open a discussion thread immediately he shouldn't have closed the thread. Closing the thread was an opportunity to cease the volume of discussion.

**Comment by krainboltgreene** [comment]
Are you going to also be suggesting edits to other posts? Is this a new service of yours? I suppose you get a lot of free time since you don't have to think about the ethical consequences of your work.

**Comment by benwiley4000** [comment]
@RyanCavanaugh:
> You can lodge this complaint in a good-faith, civil manner. For example: "Sean, I'd really appreciate it if this thread were left open until the point when the discussion thread is opened up. To me, not having an open issue for this issue means there isn't a place to discuss it, and I feel it's important that there is always an open channel for people to engage in.

I think that would be a fair rewording of my comment...

> It's been eight long hours and not having a constant unstructured discussion going at all times will certainly lead to negative outcomes, such as..."

🙄 

 As @hannahhoward mentioned in #1639:

> It is precisely the time when people are paying attention when discussion is necessary, when a wider group of perspectives can be heard.

**Comment by evocateur** [comment]
Hey, you know what's cool? Strangers on the internet assuming they know all the details of your thought process that led to a difficult decision. Super fun!

For the record, I was not pressured by anyone to revert, politically or otherwise. I do in fact have complex political beliefs, and they are decidedly on the "punch Nazis erryday" end of the spectrum.

**Comment by catcher-in-the-try** [comment]
@benwiley4000 You're running low on the benefit of the doubt when it comes to whether you're serious or a troll.

Cool down for a couple of days and come back to discuss this.

**Comment by benwiley4000** [comment]
@evocateur Daniel I don't know the details of your thought process, of course, but I do know that numerous threads were opened demanding the reverting of the license as well as the removal of Jamie from the maintainers list - to which you acquiesced, whether you felt pressured to do so or not.

I know this because in https://github.com/lerna/lerna/pull/1633 you directly cited those threads.

So on those grounds, while I'll admit your mind is a black box to me, I don't think I made any wild assumptions.

As an open source maintainer - I know this is hard, and I can appreciate you've been put in a tough position. I just cannot help thinking this could be handled much differently.

**Comment by oicorn** [comment]
First off, you don't need to assume good faith where there is none. I have seen krainboltgreene attacking people who disagrees with their politics in the Ruby community, including being insanely confrontational on a thread with Satoshi and other japanese contributors who insanely patiently explained multiple times that their politics and political orthodoxy does not match the USA's. I've seen him openly insulting both the Ruby community members and major contributors on Twitter. Your decisions should be made based on actual contributors to this project, not slacktivists showing up to dump their two cents, pressure and harass you and then leave. I see you describe yourself as matching Jaime's politics and on that topic you're seeing the true nature of some of these people who share your opinions and how they react when they don't get what they want. But I digress.

With regards to the discussion. If open source is inherently political then **whose politics determine** what politics make it into the repo? What if there is a contributor who wrote 95% of the codebase who is right wing, but 100 contributors who wrote 5% of the codebase are left wing? Do the people who contributed 5% of the project vote him off the island in order to push their own politics into the codebase of the repo? When they find out he is pushing his own politics into the software, do they remove his right to use 5% of the code, forcing him to refactor and rebase their contributions out of the codebase?

This is all unprofessional nonsense. Developers create things to be used by the masses, which in turn get used by all sorts of people and companies. I'm sure for every 1,000 people on Earth using software there are murders, thieves, adulterers and every moral failing on Earth. But you are not a criminal justice system or a church priest and it's not your job to reckon with what you deem to be moral failings. And the moment you decide you're going to start acting as judge jury and executioner for whatever political cause or moral failing you've cherrypicked, there are other people working on similar projects (and your project) who are going to disagree with you. So you schism the community between people who are on your side and people who aren't, and you make it weaker and its goals harder to achieve. Nor did you achieve your own goals, you just made people think you're a jerk. This is why politics don't belong in professional settings. Oh and by the way your own personal politics in the USA don't match other people's politics across the pond. As I mentioned above, I've seen Rubyists in Japan extremely frustrated by USA contributors trying to cram their politics down their throats.

**Comment by krainboltgreene** [comment]
I have a fan 😍

**Comment by oicorn** [comment]
Imagine being proud of being a jerk to Japanese people and making westerners look awful because you can't handle people in Japan not wanting to cram your ultra-left USA politics into their open source project

**Comment by krainboltgreene** [comment]
I didn't say you were honest, just that you were a fan.

**Comment by bkuhn** [comment]
I do believe that politics and software are definitely linked, and software freedom as a cause unto itself is inherently a political struggle for the rights of users and developers.

The question of how that specific social justice cause of software freedom should interact with other social justice causes is a complicated one, and I'm thus not surprised that this situation has caused contentious debate.  FWIW, the situation [inspired me to write down my own thoughts about the issue](https://sfconservancy.org/blog/2018/aug/30/big-tent/).

TL;DR is that I think projects should look for ways to make positive impact on other social justice issues, but using the FOSS license of the project is not productive nor in fitting with software freedom principles.

**Comment by mAAdhaTTah** [comment]
Part of problem is we can't even conceive of "politics" outside of the "left-wing/right-wing" schism, and don't realize how a comment like the below is indicative of software's inherent politics, not an argument against it.

> Developers create things to be used by the masses, which in turn get used by all sorts of people and companies.

What technology enables, both in regards to free software principles as well as user affordances, is an active decision made by the creators of software about what power we should give to our users.

And questions of power are always political.

**Comment by benwiley4000** [comment]
Beautifully stated @mAAdhaTTah.

**Comment by tbredin** [comment]
Yeah nothing involving humans can truly be apolitical; "apolitical" is almost always just coded language for "the default politics of the dominant status-quo"

**Comment by nic-hartley** [comment]
I'm genuinely surprised no one mentioned (IMO, at least) the only definite CoC violations that Jamie committed: Repeatedly calling every employee of multiple huge companies racist (I mean that [literally and directly](https://github.com/palantir/tslint/issues/4141), not just putting them on a list here, and I distinctly remember at least one other issue with the word 'racist' in the title, though I can't find it) because of who they work for. That doesn't just violate this repo's CoC, it also violates the GitHub ToS. Whether that warrants him being removed from this repo is up for debate, but I wanted to let people know that it wasn't _just_ this license change and the rampant toxicity of closing dissenting issues immediately with no reason.

**Comment by robbyoconnor** [comment]
First off -- not a lawyer. 

Secondly -- allow me to be publicly and unequivocally state that I am in no way supporting ICE or what they're doing...I am however in opposition to the methods that were used here. The license was half-baked and to top it off...I was personally emailed and had claims levied that l that I somehow support ICE because I don't support how this was executed...there are better, more thought-out ways to do this. If you truly want to make this happen, do it in a way that will actually hold up if challenged...not to mention that unless you contacted every single person who made a contribution, I'm fairly certain you can't change a license in this manner.

**Comment by robbyoconnor** [comment]
First off -- not a lawyer.

Secondly -- allow me to be publicly and unequivocally state that I am in no way supporting ICE or what they're doing...I am however in opposition to the methods that were used here. The license was half-baked and to top it off...I was personally emailed and had claims levied that l that I somehow support ICE because I don't support how this was executed...there are better, more thought-out ways to do this. If you truly want to make this happen, do it in a way that will actually hold up if challenged...not to mention that unless you contacted every single person who made a contribution, I'm fairly certain you can't change a license in this manner.

The guy emailing me seemed rather angry -- rather than doing what James Kyle and the guy who was emailing me did -- actually lobby with the leadership, not developers. Stop harassing people and actually try to change things -- and contact people who actually have the power to make such changes. You can't assume just because someone works for a company that they agree with the contracts the companies enter into.

**Comment by benwiley4000** [comment]
@robbyoconnor: Sorry you had people email harassing you - that's not something I would have called for in beginning this thread.

For the record, the MIT license modification banned use of Lerna by a list of *companies*. It did not mention use of Lerna by that companies' employees for personal projects that they own. Over and over it's been suggested that the license discriminated against company employees on an individual level, which is untrue.

**Comment by robbyoconnor** [comment]
On 08/31/2018 12:13 AM, Ben Wiley wrote:
>
> @robbyoconnor <https://github.com/robbyoconnor>: Sorry you had people
> email harassing you - that's not something I would have called for in
> beginning this thread.ize
>
Apparently in opposing this specific action in the way it played out --
I support ICE and everything they're doing? Rather than do things like
this, actually organize and thoughtfully do something. Don't harass
developers, and definitely don't harass people who didn't enter into the
contracts.
>
> For the record, the MIT license modification banned use of Lerna by a
> list of /companies/. It did not mention use of Lerna by that
> companies' employees for personal projects that they own. Over and
> over it's been suggested that the license discriminated against
> company employees on an individual level, which is untrue.
>
The problem is that it doesn't spell out enough. It was so half-baked.

**Comment by robbyoconnor** [comment]
Also @benwiley4000 with regards to me deleting my comments, apologies for that -- I realized some of my arguments were not legally sound -- not a lawyer -- so sorry about that. I tend to not leave things that I said that might not be legally sound.

**Comment by polterguy** [comment]
For the record, even though I applaud developers with a conscience, the project is licensed as MIT, which permits forking the project, and distributing it under another license, which allows for excluding any government agencies, and/or individuals.

If a bunch of its developers are determined to not work for government agencies, this (obviously) is the solution ...
